### ISSUE TITLE
TypeError: _.get is not a function

### ISSUE THREAD

**Issue by sven76** [askmike/gekko#1819]
**Note: for support questions, please join our [Discord server](https://discord.gg/26wMygt)**

* **I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Action taken** (what you did)
- Installed today's (2018-01-30) development version, updated npm stuff.
- Live Trading -> [x] Tradebot
- Exchange: Binance
- Currency: ETH
- Asset: ICX
- Strategy: UO
- API key of binance was registered before. 

* **Expected result** (what you hoped would happen)
start the trading bot

* **Actual result** (unexpected outcome)
Jan 30 20:44:29 nas node[30093]:   <-- GET /api/configPart/paperTrader
Jan 30 20:44:29 nas node[30093]:   --> GET /api/configPart/paperTrader 200 2ms 132b
Jan 30 20:44:29 nas node[30093]:   --> GET /api/strategies 200 17ms 2.07kb
Jan 30 20:45:17 nas node[30093]:   <-- POST /api/startGekko
Jan 30 20:45:17 nas node[30093]: Gekko 299484820148514 started
Jan 30 20:45:17 nas node[30093]:   --> POST /api/startGekko 200 96ms 146b
Jan 30 20:45:18 nas node[30093]: /var/lib/gekko/20180130_gekko/web/routes/startGekko.js:86
Jan 30 20:45:18 nas node[30093]:     if(_.get(event, 'log'))
Jan 30 20:45:18 nas node[30093]:          ^
Jan 30 20:45:18 nas node[30093]: TypeError: _.get is not a function
Jan 30 20:45:18 nas node[30093]:     at pipelineRunner (/var/lib/gekko/20180130_gekko/web/routes/startGekko.js:86:10)
Jan 30 20:45:18 nas node[30093]:     at Object.message (/var/lib/gekko/20180130_gekko/core/workers/pipeline/messageHandlers/realtimeHandler.js:15:9)
Jan 30 20:45:18 nas node[30093]:     at ChildProcess.<anonymous> (/var/lib/gekko/20180130_gekko/core/workers/pipeline/parent.js:29:12)
Jan 30 20:45:18 nas node[30093]:     at emitTwo (events.js:106:13)
Jan 30 20:45:18 nas node[30093]:     at ChildProcess.emit (events.js:191:7)
Jan 30 20:45:18 nas node[30093]:     at process.nextTick (internal/child_process.js:787:12)
Jan 30 20:45:18 nas node[30093]:     at _combinedTickCallback (internal/process/next_tick.js:73:7)
Jan 30 20:45:18 nas node[30093]:     at process._tickCallback (internal/process/next_tick.js:104:9)
Jan 30 20:45:18 nas systemd[1]: gekko.service: Main process exited, code=exited, status=1/FAILURE
Jan 30 20:45:18 nas systemd[1]: gekko.service: Failed with result 'exit-code'.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc)

**Comment by Belphemur** [comment]
I have the same issue. 

Just starting a watcher on Kraken make it crash.

**Comment by Bardesss** [comment]
I have the same issue, also not getting any data in a backtest.

Running Gekko on a Raspberry Pi ZeroW with Raspbian.

**Comment by askmike** [comment]
apologies, this has since been fixed here: https://github.com/askmike/gekko/commit/dd7a2759991305a4b9471538cc6ce752d21138d4 and here:  https://github.com/askmike/gekko/commit/eeddec2c2fc163cd013fdd934d8bd5f2df86ab1b

**Comment by askmike** [comment]
Some context: I obviously should not push broken code, apologies for this!

If you want to be sure to have a stable version, download a stable release from here: https://github.com/askmike/gekko/releases

These should not have these kind of problems :)

**Comment by aevrane** [comment]
_.get is still being used in the binance.js file on line 304, but what do I change it to in order to avoid this error?

**Comment by aclarke822** [comment]
@aevrane I have the same issue. If I'm understanding correctly, it looks like this happens because the .get function in Lodash only exists since 3.7.0, yet the version in the package.json is 2.x. You can't upgrade to 3.7.0 though because it will break some other functions. Here is a workaround that I'm trying: 
https://gist.github.com/jeneg/9767afdcca45601ea44930ea03e0febf
Add this function (with a more appropriate name) and replace _.get() with that function. Can't promise it will work, but I haven't had any issues. Yet.

**Comment by askmike** [comment]
@aclarke822 @aevrane @sven76 I closed this issue too early, I indeed did not remove all `_.get` instances.

I will upgrade lodash soon, but we need to update a huge part of the codebase. Almost all internal modules rely on `_.bindAll` which has been removed more recent versions of lodash.

The reason I was using _.get is because it checks whether `data` is actually an object before trying to read the `msg` property (making sure Gekko does not crash with `Cannot read property "msg" of undefined`). I fixed it manually for now, see b80db7541d479f71e65b50a2ae2c4e63730f2505.
